### PR TITLE
Adding maximize to Adamax

### PR DESCRIPTION
**Pull request by zaxtax**
Added the maximize flag #68052 to Adamax optimizer and updates the respective tests.

### CODE REVIEW

**Comment by zaxtax** [comment]
@albanD I think the failing tests have nothing to do with the contributions. I'll try to rebase against viable/strict and see if they go away.

File: torch/optim/adamax.py [hunk 1]
```diff
@@ -147,7 +153,8 @@ def adamax(params: List[Tensor],
            beta1: float,
            beta2: float,
            lr: float,
-           weight_decay: float):
+           weight_decay: float,
+           maximize: bool):
```
Comment (albanD) [inline torch/optim/adamax.py:157]: We want to make this default to False otherwise this would be a BC-breaking change.

**Comment by albanD** [comment]
Looks pretty good. Only a small detail on BC.

**Review by albanD — APPROVED** [review]
SGTM
Thanks for the PR!

**Pull request closed without merging**